src/Products/Viewer/Cache/HtmlViewer.cs

```diff
@@ -0,0 +1,215 @@
+using GroupDocs.Viewer.Results;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace GroupDocs.Viewer.MVC.Products.Viewer.Cache
+{
+    class HtmlViewer : IDisposable
+    {
+        private readonly string filePath;
+        private readonly string htmlResourcePrefix;
+        private readonly IViewerCache cache;
+
+        internal readonly GroupDocs.Viewer.Viewer viewer;
+        private readonly HtmlViewOptions viewOptions;
+        private readonly PngViewOptions pngViewOptions;
+        private readonly ViewInfoOptions viewInfoOptions;
+        private static readonly Common.Config.GlobalConfiguration globalConfiguration = new Common.Config.GlobalConfiguration();
+
+        public HtmlViewer(string filePath, string htmlResourcePrefix, IViewerCache cache, int pageNumber = -1, int newAngle = 0)
+        {
+            this.cache = cache;
+            this.filePath = filePath;
+            this.htmlResourcePrefix = htmlResourcePrefix;
+            this.viewer = new GroupDocs.Viewer.Viewer(filePath);
+            this.viewOptions = this.CreateHtmlViewOptions(pageNumber, newAngle);
+            this.pngViewOptions = this.CreatePngViewOptions(pageNumber, newAngle);
+            this.viewInfoOptions = ViewInfoOptions.FromHtmlViewOptions(this.viewOptions);
+        }
+
+        private HtmlViewOptions CreateHtmlViewOptions(int passedPageNumber = -1, int newAngle = 0)
+        {
+            HtmlViewOptions htmlViewOptions = HtmlViewOptions.ForExternalResources(
+                pageNumber =>
+            {
+                string fileName = $"p{pageNumber}.html";
+                string filePath = this.cache.GetCacheFilePath(fileName);
+
+                return File.Create(filePath);
+            },
+                (pageNumber, resource) =>
+                {
+                    string fileName = $"p{pageNumber}_{resource.FileName}";
+                    string filePath = this.cache.GetCacheFilePath(fileName);
+
+                    return File.Create(filePath);
+                },
+                (pageNumber, resource) =>
+                {
+                    var urlPrefix = "/viewer/resources/" + Path.GetFileName(this.filePath).Replace(".", "_");
+                    return $"{urlPrefix}/p{ pageNumber}_{ resource.FileName}";
+                });
+
+            htmlViewOptions.SpreadsheetOptions.TextOverflowMode = TextOverflowMode.HideText;
+            htmlViewOptions.SpreadsheetOptions.SkipEmptyColumns = true;
+            htmlViewOptions.SpreadsheetOptions.SkipEmptyRows = true;
+            SetWatermarkOptions(htmlViewOptions);
+
+            if (passedPageNumber >= 0 && newAngle != 0)
+            {
+                Rotation rotationAngle = GetRotationByAngle(newAngle);
+                htmlViewOptions.RotatePage(passedPageNumber, rotationAngle);
+            }
+
+            return htmlViewOptions;
+        }
+
+        private PngViewOptions CreatePngViewOptions(int passedPageNumber = -1, int newAngle = 0)
```

Comment:
Codacy found an issue: [Remove this unused method parameter 'passedPageNumber'.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5933705)